python/ray/data/dataset.py

```diff
@@ -345,18 +346,28 @@ def random_shuffle(self,
                 based on system randomness.
             num_blocks: The number of output blocks after the shuffle, or None
                 to retain the number of blocks.
+            move: Whether the blocks for the source dataset should be moved
```

Comment:
Let's keep this undocumented for now. We should re-think lazy vs eager execution management and come up with a public API holistically.